lib/connect/ws.js

```diff
@@ -16,9 +16,17 @@ function buildBuilder (client, opts) {
     protocol: 'mqtt'
   }
   var host = opts.hostname || 'localhost'
-  var port = String(opts.port || 80)
   var path = opts.path || '/'
-  var url = opts.protocol + '://' + host + ':' + port + path
+
+  if (!opts.port) {
+    if (opts.protocol === 'wss') {
```

Comment:
I preferred to keep the same style as for the function below, assuming it was written that way for a reason.